### PR TITLE
dom shim customElements.get

### PR DESCRIPTION
**Pull request by nojvek**

### CODE REVIEW

File: lib/isorender/dom-shims.js [hunk 1]
```diff
@@ -84,11 +84,14 @@ Document.prototype.createElement = function(tagName) {
   return el;
 };
 
-global.customElements = global.customElements || {
+global.customElements = {
```
Comment (nojvek) [inline lib/isorender/dom-shims.js]: I'm writing tests for mp-table, and it seems that environment has a globals.customElements which isn't bound to registeredElements. This makes the customElements (get, define) bound to the registry as expected.

File: lib/isorender/dom-shims.js [hunk 2]
```diff
@@ -84,11 +84,14 @@ Document.prototype.createElement = function(tagName) {
...
+  get(tagName) {
+    return registeredElements[tagName];
+  },
   define(tagName, proto) {
     tagName = tagName.toLowerCase();
     if (registeredElements[tagName]) {
-      throw DOMException(`Registration failed for type '${tagName}'. A type with that name is already registered.`);
+      throw new Error(`Registration failed for type '${tagName}'. A type with that name is already registered.`);
```
Comment (nojvek) [inline lib/isorender/dom-shims.js]: Turns out DOMException wasn't a thing. Just throwing a regular error.

File: lib/isorender/dom-shims.js [hunk 3]
```diff
@@ -84,11 +84,14 @@ Document.prototype.createElement = function(tagName) {
   return el;
 };
 
-global.customElements = global.customElements || {
```
Comment (tdumitrescu) [inline lib/isorender/dom-shims.js]: No big deal but why remove capabilities? (in this case to define/shim `customElements` elsewhere before loading this module)

File: lib/isorender/dom-shims.js [hunk 4]
```diff
@@ -84,11 +84,14 @@ Document.prototype.createElement = function(tagName) {
...
+global.customElements = {
+  get(tagName) {
+    return registeredElements[tagName];
+  },
```
Comment (tdumitrescu) [inline lib/isorender/dom-shims.js]:
```suggestion
  },

```

File: lib/isorender/dom-shims.js [hunk 5]
```diff
@@ -85,12 +85,16 @@ Document.prototype.createElement = function(tagName) {
...
+  get(tagName) {
+    return customElementsRegistry[tagName];
+  },
+
   define(tagName, proto) {
     tagName = tagName.toLowerCase();
-    if (registeredElements[tagName]) {
-      throw DOMException(`Registration failed for type '${tagName}'. A type with that name is already registered.`);
+    if (customElementsRegistry[tagName]) {
+      throw new Error(`CustomElementRegistryError: '${tagName}' name has already been registered`);
```
Comment (tdumitrescu) [inline lib/isorender/dom-shims.js]: Out of curiosity, where does this message come from? The previous one came from Chrome's implementation at the time I think.. Current Chrome is
```
"Failed to execute 'define' on 'CustomElementRegistry': this name has already been used with this registry
    at <location>>"
```

File: lib/isorender/dom-shims.js (hunk 3 above)
Comment (nojvek) [reply lib/isorender/dom-shims.js]: Put it back, made the customElement registry not override itself. It seems mochapack is somehow initializing the module twice. So much state :(

File: test/server/dom-shims.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import '../../lib/isorender/dom-shims';
+
+import {expect} from 'chai';
+
+import {SimpleApp} from '../fixtures/simple-app';
+
+customElements.define(`my-app`, SimpleApp);
+
+describe(`customElements registry`, function() {
+  it(`.get() returns component class`, function() {
```
Comment (tdumitrescu) [inline test/server/dom-shims.js]:
```js
  describe(`.get()`, function() {
    it(`returns component class`, function() {
      // etc
    });
  });
```

File: lib/isorender/dom-shims.js (hunk 3 above)
Comment (tdumitrescu) [reply lib/isorender/dom-shims.js]: smh for mochapack

File: lib/isorender/dom-shims.js (hunk 5 above)
Comment (nojvek) [reply lib/isorender/dom-shims.js]: Nah just a rewording of the chrome error. Can make it exactly the same if you want.

File: test/server/dom-shims.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/* eslint-env mocha */
```
Comment (tdumitrescu) [inline test/server/dom-shims.js]:
```suggestion
/* eslint-env mocha */

```

File: test/server/dom-shims.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import {expect} from 'chai';
+
+import {SimpleApp} from '../fixtures/simple-app';
+
+customElements.define(`my-app`, SimpleApp);
+
+describe(`customElements registry`, function() {
+  describe(`.get()`, function() {
+    it(`returns component class`, function() {
+      expect(customElements.get(`my-app`)).to.eql(SimpleApp);
```
Comment (tdumitrescu) [inline test/server/dom-shims.js]: strict equality should work here, right?
```suggestion
      expect(customElements.get(`my-app`)).to.equal(SimpleApp);
```

File: test/server/dom-shims.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    });
+  });
+
+  describe(`.define()`, function() {
+    it(`called twice throws an error`, function() {
```
Comment (tdumitrescu) [inline test/server/dom-shims.js]:
```suggestion
    it(`throws an error if called twice with the same tag name`, function() {
```

File: test/server/dom-shims.js (hunk 8 above)
Comment (nojvek) [reply test/server/dom-shims.js]: oh is that what the difference between .eql and .equal was. I assumed .eql was an alias. #TIL

Comment (tdumitrescu) [reply test/server/dom-shims.js]: `eql` in chai is deep-equals, but afaik we can rely on comparing these by reference

**Pull request merged**